apis/telemetry/v1alpha1/telemetry_types.go

```diff
@@ -15,6 +20,32 @@ type PrometheusTelemetrySpec struct {
 	CommonLabels map[string]string `json:"commonLabels,omitempty"`
 }
 
+type VectorSpec struct {
+	// Enabled enables the Vector agent for this resource (Cassandra, Reaper or Stargate).
+	// Enabling the vector agent will inject a sidecar container into the pod.
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// Config is the name of the configmap containing custom sinks and transformers for the Vector agent.
+	// The configmap must be in the same namespace as the CassandraDatacenter and contain a vector.toml entry
+	// with the Vector configuration in toml format.
+	// The agent is already configured with a "cassandra_metrics" source that needs to be used as input for the sinks.
+	// If not set, the default console sink will be used.
+	Config *corev1.LocalObjectReference `json:"config,omitempty"`
+
+	// Resources is the resource requirements for the Vector agent.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Image is the name of the Vector image to use. If not set, the default image will be used.
+	// +optional
+	// kube:default="timberio/vector:0.26.0-alpine"
+	Image string `json:"image,omitempty"`
+
+	// ScrapeInterval is the interval at which the Vector agent will scrape the metrics endpoint.
```

Comment:
Would be nice to specify which time unit to use (I imagine seconds?).

Reply:
Also, in meta/v1 there is a Duration type:

```go
// Duration is a wrapper around time.Duration which supports correct
// marshaling to YAML and JSON. In particular, it marshals into strings, which
// can be used as map keys in json.
type Duration struct {
	time.Duration `protobuf:"varint,1,opt,name=duration,casttype=time.Duration"`
}
```

Reply:
Sounds good, I'll use the Duration type.